tests/handlers/test_register.py

```diff
@@ -235,3 +236,20 @@ def test_invalid_user_id_length(self):
             self.handler.register(localpart=invalid_user_id),
             SynapseError
         )
+
+    def test_email_to_displayname_mapping(self):
+        """Test that custom emails are mapped to new user displaynames correctly"""
+        i = "jean-philippe.martin-laval@developpement-durable.fr"
+        expected = "Jean-Philippe Martin-Laval [Developpement-Durable]"
+        result = _map_email_to_displayname(i)
+        assert result == expected, "%s != %s" % (i, expected)
```

Comment:
```suggestion
        self.assertEqual(result, expected)
```

is probably what you want